### PR TITLE
ocm: use separate dir in jumphost for each dag run

### PR DESCRIPTION
**Pull request by venkataanil**
1. Dedicate airflow-ocm-jumphost.rdu2.scalelab.redhat.com ORCHESTRATION_HOST for only prod
2. Run each end point testing individually (to avoid one test failure impacting another).
3. Use /tmp/{UUID} folder for the run, so each run will have dedicated run directory
4. Before running the test check and create aws keys for OsdCcsAdmin user
5. Declare all tests with their rate & duration inside the script itself instead of passing as parameters to avoid complexity.
6. Add cleanup script to delete stale resources (i.e if ocm-api-load failed to delete them)
7. Run each test with timeout. This allows ocm-api-load to cleanup resources though it is killed by airflow for any reason. This also avoids ocm-api-load hanging
8. Avoid service tests till they get stable. Avoid create_cluster test as it is not working with cli args

### CODE REVIEW

File: dags/nocp/scripts/run_ocm_api_load.sh [hunk 1]
```diff
@@ -1,77 +1,113 @@
...
+get-current-account 10/s 10 \n
+quota-cost 10/s 30 \n
+resource-review 5/s 30 \n
+cluster-authorizations 10/s 15 \n
+self-terms-review 30/s 10 \n
+certificates 15/s 10"
+
+create_aws_key(){
+    # Delete aws keys if more than 1 key exists
+    arr=(`/usr/bin/aws iam list-access-keys --user-name OsdCcsAdmin --output text --query 'AccessKeyMetadata[*].AccessKeyId'`)
```
[the remark this answers is not in the document]
Comment (venkataanil) [reply dags/nocp/scripts/run_ocm_api_load.sh:21]: ack

**Comment by venkataanil** [comment]
I  have added README.md which explains all the modules.

File: dags/nocp/README.md [hunk 2]
```diff
@@ -1,14 +1,57 @@
...
-* `util`: Contains utility functions for Airflow
\ No newline at end of file
+* `util`: Contains utility functions for Airflow
+
+# OCM DAG
+
+This DAG is used for OCM testing. Modules in this DAG -
+
+*1) config/benchmarks/ocm.json*
+* Secrets are fetched from vault.  All config variables defined in this file
```
Comment (krishvoor) [inline dags/nocp/README.md:25]: Minot nit:
`Secrets are fetched from the vault.`

File: dags/nocp/README.md [hunk 3]
```diff
@@ -1,14 +1,57 @@
...
+
+*2) manifest.yaml*
+* This manifest file defines ocm app and its auto schedule configuration (every Thursday 9am UTC). Autoschedule is enabled for only prod dag
```
Comment (krishvoor) [inline dags/nocp/README.md:28]: Minor nit:
`Auto schedule is only enabled for the Prod. dag`

File: dags/nocp/README.md [hunk 4]
```diff
@@ -1,14 +1,57 @@
...
+
+*3) tasks/benchmarks/nocp.py*
+* OCM tasks module. It defines “ocm-api-load” benchmark and “cleanup” tasks.
+* Ocm-api-load task executes run_ocm_benchmark.sh with “ocm-api-load” operation
+* cleanup task executes run_ocm_benchmark.sh with “cleanup” operation
```
Comment (krishvoor) [inline dags/nocp/README.md:33]: Defines the **ocm-api-load** benchmark and **cleanup** tasks.
* ocm-api-load task executes `run_ocm_benchmark.sh` with `ocm-api-load` operation
* cleanup task executes `run_ocm_benchmark.sh` with `cleanup` operation

File: dags/nocp/README.md [hunk 5]
```diff
@@ -1,14 +1,57 @@
...
+
+*4) scripts/run_ocm_benchmark.sh*
+* Creates a tmp directory in the orchestration host and uses it to run the ocm-api-load or cleanup scripts.
+* Copies environment variables to a file in this tmp directory.
+* Each dag run uses its own tmp directory to assist in simultaneous dag runs.
+* Also both ocm-api-load and cleanup tasks use separate tmp directories to avoid failures of ocm-api-load impacting cleanup task.
```
Comment (krishvoor) [inline dags/nocp/README.md:39]: `s/tmp/TEMP/g`

File: dags/nocp/README.md [hunk 6]
```diff
@@ -1,14 +1,57 @@
...
+* UUID of ocm-api-load is passed to the cleanup task for removing ocm-api-load config files.
+
+*5) scripts/run_ocm_api_load.sh*
+* Defines ocm-api-load tests (end points) with rate and duration. Each test is triggered with this rate and for the specified duration
+* Some tests require OsdCcsAdmin user’s aws key. Before running the test we create this key.
```
Comment (krishvoor) [inline dags/nocp/README.md:44]: Some tests require `OsdCcsAdmin` privileges  (which are regenerated every time) before the test is started.

File: dags/nocp/README.md [hunk 7]
```diff
@@ -1,14 +1,57 @@
...
+* Use the unique tmp directory (created by run_ocm_benchmark.sh) inside the orchestration host to run the tests.
+* Clone ocm-api-load and build the ocm-load-test binary
```
Comment (krishvoor) [inline dags/nocp/README.md:46]: Clones the `ocm-api-load` repo. and build the `ocm-load-test` binary

File: dags/nocp/README.md [hunk 8]
```diff
@@ -1,14 +1,57 @@
...
+* Each test is called with a timeout.
+  Timeout = test duration + 10 minutes.
```
Comment (krishvoor) [inline dags/nocp/README.md:48]: `Timeout = test_duration + 10 minutes`

Comment (krishvoor) [reply dags/nocp/README.md:48]: Wrap it in code style format

File: dags/nocp/README.md [hunk 9]
```diff
@@ -1,14 +1,57 @@
...
+  This extra 10 minutes help test to create necessary result files after running the test for given duration
+* Automation.py uploads all result files to snappy server
```
Comment (krishvoor) [inline dags/nocp/README.md:50]: Wrap it in code style format

**Review by krishvoor — APPROVED** [review]
Pretty minor nits, lgtm overall.

**Review by krishvoor — APPROVED** [review]
lgtm

**Review by chentex — APPROVED** [review]
Wow this looks like it took a lot to put together, great work!

File: dags/nocp/config/benchmarks/ocm.json [hunk 10]
```diff
@@ -5,38 +5,11 @@
...
-		"CERTIFICATES_RATE": "15/s",
-		"CERTIFICATES_DURATION": "10",
-		"CREATE_SERVICES_RATE": "1/m",
-		"CREATE_SERVICES_DURATION": "1",
-		"GET_SERVICES_RATE": "1/m",
-		"GET_SERVICES_DURATION": "1",
-		"PATCH_SERVICES_RATE": "1/m",
-		"PATCH_SERVICES_DURATION": "1",
+		"ORCHESTRATION_USER": "root",
+		"ORCHESTRATION_HOST": "airflow-ocm-jumphost.rdu2.scalelab.redhat.com",
```
Comment (dry923) [inline dags/nocp/config/benchmarks/ocm.json:12]: Can this be a vault var instead of being hard coded into our repo?

Comment (venkataanil) [reply dags/nocp/config/benchmarks/ocm.json]: thanks Russ, addressed it.

**Review by dry923 — APPROVED** [review]
LGTM!

**Pull request merged**